src/components/badge/badge.component.tsx

```diff
@@ -15,21 +18,52 @@ export interface BadgeProps {
   onClick?: (ev: React.MouseEvent<HTMLElement>) => void;
 }
 
-export const Badge = ({ children, counter = 0, onClick }: BadgeProps) => {
-  return (
-    <StyledBadgeWrapper>
-      {counter > 0 && (
+export const Badge = ({
+  "aria-label": ariaLabel,
+  children,
+  counter = 0,
+  onClick,
+}: BadgeProps) => {
+  const shouldDisplayCounter = counter > 0;
+  const displayOnly = !onClick && shouldDisplayCounter;
+  const counterToDisplay = counter > 99 ? 99 : counter;
+
+  const renderCorrectBadge = () => {
+    if (displayOnly) {
```

Comment:
You can conditionally spread the props here if you went with the polymorphic approach I suggested below (using `as` prop)

```
const props = onClick ? {
      buttonType: "secondary",
      onClick
    } : {
      "aria-label": ariaLabel
    }
    
    if (shouldDisplayCounter) {
      return (
        <StyledBadge
          data-component="badge"
          {...props}
        >
          {onClick && <StyledCrossIcon data-element="badge-cross-icon" type="cross" />}
          <StyledCounter data-element="badge-counter">
            {counterToDisplay}
          </StyledCounter>
        </StyledBadge>
      );
    }
 ```